### ISSUE TITLE
Size mismatches and missing keys with testing

### ISSUE THREAD

**Issue by mathemmatics** [cszn/KAIR#24]
Hi, so after I've trained for a bit I want to actually use this model in another code. So following the example set in main_test, I run the following snippet:

        denoiser = net(in_nc=1, out_nc=1, nc=64, nb=17, act_mode='R')
        denoiser.load_state_dict(torch.load(os.path.join(args.model_dir, args.model_name)), strict=True)
        denoiser.eval()

However when it loads in the state dict, I get a myriad of size mismatches and missing keys. Admittedly I am terminating training early because its currently taking ~10 days to train a network, but I don't understand why it's able to run this model in the training mode but not in testing mode.

**Comment by NJXZhang** [comment]
I found the problem in `main_test_dncnn.py`. The issue is that `act_mode` in `main_test_dncnn.py` is hardcoded. Wondering if your issue is related to that.

**Comment by cszn** [comment]
https://github.com/cszn/KAIR/blob/405f3c0c8fd0958ae3bff113f864c0a57a5b70db/main_test_dncnn.py#L119
Try to set `act_mode='BR'`.

**Comment by mathemmatics** [comment]
Yep, that works! Thank you so much.

**Comment by mathemmatics** [comment]
I'm now running into an issue when the epochs is greater than 400000. Is there another special mode that I should be aware of to get this to run?

**Comment by cszn** [comment]
Two possible solutions:
1) Set "merge_bn": False, https://github.com/cszn/KAIR/blob/6ac53385f707ca4c0e7242ce62ea123cea2a9d39/options/train_dncnn.json#L12
2) Set act_mode='R'

**Comment by mathemmatics** [comment]
I've tried solution 2 in my testing, but still inoperable.
